mars-rover/mars_rover.rb

```diff
@@ -1,3 +1,151 @@
+# class Dimensions is to define the maximum size of the map (N, E, S, W)
+class Dimensions
+
+   def initialize (n,e,s,w)
+     $max_n = n
+	 $max_e = e
+	 $max_s = s
+	 $max_w = w
+   end
+   
+   def get_dimensions
+      return [$max_n,$max_e,$max_s,$max_w]
+   end
+ end
+
+# class Obstacle is to define a position of an obstacle on the map
+class Obstacle
+
+  def initialize (x,y)
+   $obstacle_pos_x = x
+   $obstacle_pos_y = y
+  end
+
+  def get_obstacle_position
+      return [$obstacle_pos_x,$obstacle_pos_y]
+  end
+end
+
+# class MarsRover is to move the MarsRover on the map taking into account the position of an obstacle and the dimensions of the map.
 class MarsRover
+  DIRECTIONS = ['N','E','S','W']
+
+  def initialize (x,y,direction)
+   @pos_x = x
+   @pos_y = y
+   @direction = DIRECTIONS.index(direction)
+   @commands = []
+  end
 
+  def get_position
+      return [@pos_x,@pos_y,DIRECTIONS[@direction]]
+  end
+
+  def move rovercommand
+  
+# Get the commands and put them in an array
```

Comment:
Commenting is good, but try to focus on the 'why', and let your code be the 'what'